lib/helper.js

```diff
@@ -24,6 +24,10 @@ function encodeKey(metricName, tags) {
   }
   // Sort the tags based on their keys to ensure we don't duplicate metrics if the same tags come out of order.
   var tagsArray = Object.keys(tags).sort().map(key => [key, tags[key]]);
+  var tagsType = typeof tags;
+  if (tagsType != "object") {
+    throw "Wrong Tags Object Sent To The API"
```

Comment:
We can make this exception as clear to the user as possible with info about "why it failed" and "the expected format"